lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/RequestFactory.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Client.YARN.RestClient
+{
+    /// <summary>
+    /// Factory to generate REST requests
+    /// </summary>
+    internal class RequestFactory : IRequestFactory
+    {
+        private readonly ISerializer _serializer;
+        private readonly string _baseResourceString;
```

Comment:
This will remain the same for a particular version of the API. Since we are not adding handling for API version changes yet, we can assume that this will not change (in the current form of the code).
Stated differently, if this changes a lot more stuff has to change.